### ISSUE TITLE
IWorldElement.Parent and FindNearestParent<T> ProtoFlux Nodes

### ISSUE THREAD

**Issue by art0007i** [Yellow-Dog-Man/Resonite-Issues#609]
### Is your feature request related to a problem? Please describe.

If I have a reference to a field in protoflux, there is no way to find out what Component or Slot that field belongs to.

### Describe the solution you'd like

A `FindNearestParent<T>` node which would take in an `IWorldElement` and find the first parent of type T
for example if I plug in a `Sync<bool>` into a `FindNearestParent<Slot>` it would get the parent of `Sync<bool>` which could be some component, then it would get the parent of the component which is a slot, and return that.

basically this is just a call into the `WorldElementExtensions.FindNearestParent<T>` function

The `IWorldElement.Parent` node would just get the parent of the world element. Using this node it would be possible to build the FindNearestParent node manually.

### Describe alternatives you've considered

n/a

**Comment by badhaloninja** [comment]
Does the get slot node not work with fields?

**Comment by Frooxius** [comment]
We'll probably introduce these after collections are implemented and we work on a large set of component access nodes. These would come as part of that.
